src/auth/jwtclient.ts

```diff
@@ -150,6 +150,21 @@ export class JWT extends OAuth2Client {
     }
   }
 
+  /**
+   * @param targetAudience the audience for the fetched ID token.
+   * @return URL to consent page.
+   */
+  getIdTokenClient(targetAudience: string): OAuth2Client {
+    return new JWT({
```

Comment:
would we need to implement `fetchIdToken` in `jwtclient` as well?

Reply:
We could! this is already happening on line 123:

```js
if (
  this.additionalClaims &&
  (this.additionalClaims as {
    target_audience: string;
  }).target_audience
) {
  const {tokens} = await this.refreshToken();
  return {
    headers: this.addSharedMetadataHeaders({
      Authorization: `Bearer ${tokens.id_token}`,
    }),
  };
}
```
I could peel that part out, but since it's basically the same I didn't see much of a point.

Reply:
Oh interesting, I just noticed that `fetchIdToken` is private on ComputeClient, ignore me. If I'm following correctly:

1. a user instantiates their auth client with `targetAudience`, or for `JWT`, with `{additionalClaims: target_audience}`.
1. requests made will have additional headers set, signed with keys corresponding to public keys.
1. an upstream service can use `verify` to ensure that the request being made was from the appropriate audience, based on these same public keys?

---

I think for consistency we should probably make `JWT` accept `targetAudience` in the constructor, and just have it be an alias for `{target_audience: targetAudience}`, maybe? so that the constructor is the same as the `Compute` instance?